src/terminal/parser/ut_parser/OutputEngineTest.cpp

```diff
@@ -1805,215 +1672,56 @@ class StateMachineExternalTest final
 
         pDispatch->ClearState();
     }
 
-    TEST_METHOD(TestAutoWrapMode)
+    TEST_METHOD(TestPrivateModes)
     {
-        auto dispatch = std::make_unique<StatefulDispatch>();
-        auto pDispatch = dispatch.get();
-        auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
-        StateMachine mach(std::move(engine));
-
-        mach.ProcessString(L"\x1b[?7l");
-        VERIFY_IS_FALSE(pDispatch->_isAutoWrapEnabled);
-
-        pDispatch->ClearState();
-        pDispatch->_isAutoWrapEnabled = false;
-
-        mach.ProcessString(L"\x1b[?7h");
-        VERIFY_IS_TRUE(pDispatch->_isAutoWrapEnabled);
-
-        pDispatch->ClearState();
-    }
-
-    TEST_METHOD(TestCursorBlinking)
-    {
-        auto dispatch = std::make_unique<StatefulDispatch>();
-        auto pDispatch = dispatch.get();
-        auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
-        StateMachine mach(std::move(engine));
-
-        mach.ProcessString(L"\x1b[?12h");
-        VERIFY_IS_TRUE(pDispatch->_cursorBlinking);
-
-        pDispatch->ClearState();
-
-        mach.ProcessString(L"\x1b[?12l");
-        VERIFY_IS_FALSE(pDispatch->_cursorBlinking);
-
-        pDispatch->ClearState();
-    }
-
-    TEST_METHOD(TestCursorVisibility)
-    {
-        auto dispatch = std::make_unique<StatefulDispatch>();
-        auto pDispatch = dispatch.get();
-        auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
-        StateMachine mach(std::move(engine));
-
-        mach.ProcessString(L"\x1b[?25h");
-        VERIFY_IS_TRUE(pDispatch->_cursorVisible);
-
-        pDispatch->ClearState();
-
-        mach.ProcessString(L"\x1b[?25l");
-        VERIFY_IS_FALSE(pDispatch->_cursorVisible);
-
-        pDispatch->ClearState();
-    }
-
-    TEST_METHOD(TestAltBufferSwapping)
-    {
-        auto dispatch = std::make_unique<StatefulDispatch>();
-        auto pDispatch = dispatch.get();
-        auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
-        StateMachine mach(std::move(engine));
-
-        mach.ProcessString(L"\x1b[?1049h");
-        VERIFY_IS_TRUE(pDispatch->_isAltBuffer);
-
-        pDispatch->ClearState();
-
-        mach.ProcessString(L"\x1b[?1049h");
-        VERIFY_IS_TRUE(pDispatch->_isAltBuffer);
-        mach.ProcessString(L"\x1b[?1049h");
-        VERIFY_IS_TRUE(pDispatch->_isAltBuffer);
-
-        pDispatch->ClearState();
-
-        mach.ProcessString(L"\x1b[?1049l");
-        VERIFY_IS_FALSE(pDispatch->_isAltBuffer);
-
-        pDispatch->ClearState();
-
-        mach.ProcessString(L"\x1b[?1049h");
-        VERIFY_IS_TRUE(pDispatch->_isAltBuffer);
-        mach.ProcessString(L"\x1b[?1049l");
-        VERIFY_IS_FALSE(pDispatch->_isAltBuffer);
-
-        pDispatch->ClearState();
+        BEGIN_TEST_METHOD_PROPERTIES()
+            TEST_METHOD_PROPERTY(L"Data:modeNumber", L"{1, 3, 5, 6, 7, 12, 25, 40, 1049}")
+        END_TEST_METHOD_PROPERTIES()
```

Comment:
This is a replacement for all the individual mode testing methods which were checking whether a particular mode change would trigger a call to a one of the mode setting methods in `ITermDispatch` (which have now been removed). But the code path we were testing wasn't real - it was just the `StatefulDispatch` mock class above, so it was kind of pointless. This test now just confirms that the `_ModeParamsHelper` method is triggered, which is all we really care about here.

Reply:
HA. Thank you!